Libraries/LibWeb/Page/EditEventHandler.cpp

```diff
@@ -120,9 +100,12 @@ void EditEventHandler::handle_insert(DOM::Position position, u32 code_point)
         builder.append(node.data().substring_view(position.offset()));
         node.set_data(builder.to_string());
 
-        m_frame.cursor_position().set_offset(m_frame.cursor_position().offset() + 1);
         node.invalidate_style();
     }
+
+    // FIXME: We need to remove stale layout nodes when nodes are removed from the DOM. Currently,
+    //        this is the only way to get these to disappear.
```

Comment:
This is the correct thing to do IMO. Trying to get fancy with this will lead to unpleasant levels of complexity.

Reply:
This is on the entire document, not just that sub-tree. In my understanding, the correct behavior of `node.invalidate_style` would be to recompute the layout for that sub-tree at the next opportune moment.